code/script.js

```diff
@@ -225,101 +285,175 @@ const generateBoard = () => {
+  console.log(`secret character is ${secret.name}`)
+}
+//This function shows the number of guesses (later invokes in filter button event listener)
+//Also the DOM selectors inside function because i want it to retrieve the updated count!
+const guessCountDisplay = () => {
+  const guessCountBox = document.getElementById('guess-container');
+  const winOrLoseGuessContainer = document.getElementById('win-or-lose-guess-container');
+
+  guessCountBox.innerText = `${guessCount}`;
+  winOrLoseGuessContainer.innerText = `${guessCount}`;
+}
+
+//This function starts the timer!
+const startTimer = () => {
+  timerInterval = setInterval(() => {
+    seconds++;
+    timerElement.innerText = seconds; 
+  }, 1000); // updates the timer every second!
+}
+//This function stops the timer! And gives the elapsed time!
+const stopTimer = () => {
+  clearInterval(timerInterval);
+  return seconds;
 }
 
+
 // This function to start (and restart) the game
 const start = () => {
+
   // Here we're setting charactersInPlay array to be all the characters to start with
   charactersInPlay = CHARACTERS
-  // What else should happen when we start the game?
+  generateBoard();
+  setSecret(); // secret character is set
 }
 
 // setting the currentQuestion object when you select something in the dropdown
 const selectQuestion = () => {
   const category = questions.options[questions.selectedIndex].parentNode.label
+  const value = questions.options[questions.selectedIndex].value
 
-  // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
-
+  //Saving the values from above for next function
   currentQuestion = {
     category: category,
-    // value: value
+    value: value,
   }
+  console.log("question selected", currentQuestion);
 }
 
 // This function should be invoked when you click on 'Find Out' button.
 const checkQuestion = () => {
   const { category, value } = currentQuestion
-
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
+ 
   if (category === 'hair' || category === 'eyes') {
-
+    //to see that value is the same, then keep is loaded with true, invoke next func  :)
+   if (secret[category] === value) { 
+    keep = true
+    filterCharacters(true);
+   } //to see that value is NOT same, then keep is loaded with false, invoke next func  :)
+   else {
+    keep = false
+    filterCharacters(false);
+   }
   } else if (category === 'accessories' || category === 'other') {
-
+    //to se if object has any value set at all for that question, true of false. 
+    if (secret[category].includes(value)) {
+      keep = true
+      filterCharacters(true);
+    } else {
+      keep = false
+      filterCharacters(false);
+    }
   }
 }
```

Comment:
Nice if/else, short and concise, makes the code look clean and the comments help the reader understand what is going on 👊🏻